### PR TITLE
Gravity Wiz // Gravity Forms // Validate Current Password

### PR DESCRIPTION
**Pull request by sbassah**
Original Gist: https://gist.github.com/spivurno/eebad70856ec3be400ccdeface37479a

### CODE REVIEW

File: gravity-forms/gw-validate-current-password.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?php
+/**
+ * Gravity Wiz // Gravity Forms // Validate Current Password
+ * http://gravitywiz.com/
+ */
+// update "123" to your form ID, update "6" to your Password field ID
```
Comment (spivurno) [inline gravity-forms/gw-validate-current-password.php:6]: Update this line to read:
```
// Update "123" to your form ID and "4" to your Password field ID.
```

File: gravity-forms/gw-validate-current-password.php [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+add_filter( 'gform_field_validation_1639_1', function( $result, $value ) {
```
Comment (spivurno) [inline gravity-forms/gw-validate-current-password.php:7]: Update "1639" here to "123" and "1" to "4".

File: gravity-forms/gw-validate-current-password.php [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+	$user = wp_get_current_user();
+	if ( ! wp_check_password( $value, $user->data->user_pass, $user->ID ) ) {
```
Comment (spivurno) [inline gravity-forms/gw-validate-current-password.php:9]: Add a check to see if there the the `$user` variable is truthy.

**Review by spivurno — CHANGES_REQUESTED** [review]

File: gravity-forms/gw-validate-current-password.php (hunk 1 above)
Comment (sbassah) [reply gravity-forms/gw-validate-current-password.php]: Done

File: gravity-forms/gw-validate-current-password.php (hunk 2 above)
Comment (sbassah) [reply gravity-forms/gw-validate-current-password.php]: Done

File: gravity-forms/gw-validate-current-password.php (hunk 3 above)
Comment (sbassah) [reply gravity-forms/gw-validate-current-password.php]: @spivurno I've added this check

File: gravity-forms/gw-validate-current-password.php [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+ */
+// update "123" to your form ID, update "4" to your Password field ID
+add_filter( 'gform_field_validation_123_4', function( $result, $value ) {
+	$user = wp_get_current_user();
+	if ($user){
+		if ( ! wp_check_password( $value, $user->data->user_pass, $user->ID ) ) {
+			$result['is_valid'] = false;
+			$result['message'] = 'Invalid current password. Please try again.';
+		}
+	}
```
Comment (spivurno) [inline gravity-forms/gw-validate-current-password.php:14]: Let's simplify this to a single line check.
```
if ( ! $user || ! wp_check_password( $value, $user->data->user_pass, $user->ID ) ) { 
```

**Review by spivurno — CHANGES_REQUESTED** [review]

File: gravity-forms/gw-validate-current-password.php (hunk 1 above)
Comment (spivurno) [reply gravity-forms/gw-validate-current-password.php]: I'm not seeing this change?

Comment (spivurno) [reply gravity-forms/gw-validate-current-password.php]: Oh, it looks like you changed the IDs but not the punctutation. You can just copy and paste the exact string I provided. 👌

Comment (sbassah) [reply gravity-forms/gw-validate-current-password.php]: Sorry about that. Guess I was focused on the IDs

File: gravity-forms/gw-validate-current-password.php (hunk 4 above)
Comment (sbassah) [reply gravity-forms/gw-validate-current-password.php]: I just updated it to a single line check.

Comment (spivurno) [reply gravity-forms/gw-validate-current-password.php]: Haha, don't hate me but I gave you some bad logic. 😆

This should only happen if the user is logged in and the password is bad.

```
if ( $user && ! wp_check_password( $value, $user->data->user_pass, $user->ID ) ) { 
```

**Review by spivurno — APPROVED** [review]
One change in the logic (my bad!) and this is good to go!

File: gravity-forms/gw-validate-current-password.php (hunk 4 above)
Comment (sbassah) [reply gravity-forms/gw-validate-current-password.php]: Yes, you're right. That fixes it.

**Pull request merged**